lib/internal/test_runner/runner.js

```diff
@@ -319,7 +349,21 @@ class FileTest extends Test {
 
 function runTestFile(path, filesWatcher, opts) {
   const watchMode = filesWatcher != null;
-  const subtest = opts.root.createSubtest(FileTest, path, { __proto__: null, signal: opts.signal }, async (t) => {
+  const Factory = opts.isolation === 'none' ? InProcessFileTest : SpawnFileTest;
+  const subtest = opts.root.createSubtest(Factory, path, { __proto__: null, signal: opts.signal }, async (t) => {
+    if (opts.isolation === 'none') {
+      let parentURL;
+      try {
+        parentURL = pathToFileURL(process.cwd() + '/').href;
+      } catch {
+        parentURL = 'file:///';
+      }
+
+      const { esmLoader } = require('internal/process/esm_loader');
+
+      await esmLoader.import(isAbsolute(path) ? path : `./${path}`, parentURL, { __proto__: null });
```

Comment:
It's unsafe to pass a path to an API that expects a URL, if the path contains any URL special char (`?`, `#`, `%`) it might be interpreted completely differently that what you meant.
```suggestion
      await esmLoader.import(pathToFileURL(path), parentURL, { __proto__: null });
```